deftree/deftree.go

```diff
@@ -136,23 +136,36 @@ func (self *MicroserviceDefinition) GetByName(name string) Describable {
 	return nil
 }
 
+type notFoundErr struct {
```

Comment:
@zaquestion  @hasIan In testing this out I found that doing this does not let us discern between different errors. For example, see the following file:
```
// Demonstration that these all appear as the all types according to type
// assertions, as they all fulfill interfaces which are equivalent. So there's
// no way to differentiate.
package main

import "fmt"

type thing struct {
	err error
}

func (t thing) Error() string {
	return t.err.Error()
}

type secondThing error

func main() {
	x := func() error {
		return thing{fmt.Errorf("test")}
	}()
	y := func() error {
		return secondThing(fmt.Errorf("test"))
	}()
	_, ok1 := x.(error)
	fmt.Printf("x is error: %v\n", ok1)
	_, ok2 := x.(thing)
	fmt.Printf("x is thing: %v\n", ok2)
	_, ok3 := x.(secondThing)
	fmt.Printf("x is secondThing: %v\n", ok3)

	_, ok1 = y.(error)
	fmt.Printf("y is error: %v\n", ok1)
	_, ok2 = y.(secondThing)
	fmt.Printf("y is secondThing: %v\n", ok2)
	_, ok3 = y.(secondThing)
	fmt.Printf("y is secondThing: %v\n", ok3)
}
```
If I run this, you can see that all variables appear to be all types at the same time:

```$ go run test.go 
x is error: true
x is thing: true
x is secondThing: true
y is error: true
y is secondThing: true
y is secondThing: true
```
-----
As far as I can tell, the only options remaining which I know do work are:
1. Go back to optional interfaces
2. Use string comparisons